packages/shim/src/lib/shim.ts

```diff
@@ -0,0 +1,340 @@
+    return this.getDomainStackTop() || this.getProcessStackTop();
+  }
+
+  /**
+   * TODO
+   */
+  public isOlderThan(version: number): boolean {
+    return API_VERSION < version;
+  }
+
+  /**
+   * TODO
+   */
+  public pushScope(client?: any): void {
+    const usedClient = client || getCurrentClient();
+    const layer: ScopeLayer = {
+      client: usedClient,
+      scope: usedClient.getInitialScope(),
+      type: 'local',
+    };
+    const stack = this.getDomainStack();
+    if (stack !== undefined) {
+      stack.push(layer);
+    } else {
+      this.getProcessStack().push(layer);
+    }
+  }
+
+  /**
+   * TODO
+   */
+  public popScope(): boolean {
+    const stack = this.getDomainStack();
+    if (stack !== undefined) {
+      return stack.pop() !== undefined;
+    }
+    return this.getProcessStack().pop() !== undefined;
+  }
+
+  /**
+   * TODO
+   */
+  public withScope(arg1: (() => void) | any, arg2?: (() => void) | any): void {
+    let client: any = arg2;
+    let callback: () => void = arg1;
+    if (!!(arg1 && arg1.constructor && arg1.call && arg1.apply)) {
+      callback = arg1;
+      client = arg2;
+    }
+    if (!!(arg2 && arg2.constructor && arg2.call && arg2.apply)) {
+      callback = arg2;
+      client = arg1;
+    }
+    this.pushScope(client);
+    try {
+      callback();
+    } finally {
+      this.popScope();
+    }
+  }
+
+  /**
+   * TODO
+   */
+  public clearScope(): void {
+    const top = this.getStackTop();
+    top.scope = top.client.getInitialScope();
+  }
+
+  /**
+   * TODO
+   */
+  private getProcessStack(): ScopeLayer[] {
+    return global.__SENTRY__.processStack;
+  }
+
+  /**
+   * TODO
+   */
+  private getProcessStackTop(): ScopeLayer {
+    const stack = this.getProcessStack();
+    if (stack.length === 0) {
+      stack.push({
+        scope: {},
+        type: 'process',
+      });
+    }
+    return stack[stack.length - 1];
+  }
+
+  /**
+   * TODO
+   */
+  private getDomainStack(): ScopeLayer[] | undefined {
+    if (!shimDomain.active) {
+      return undefined;
+    }
+    let sentry = shimDomain.active.__SENTRY__;
+    if (!sentry) {
+      shimDomain.active.__SENTRY__ = sentry = { domainStack: [] };
+    }
+    return sentry.domainStack;
+  }
+
+  /**
+   * TODO
+   */
+  private getDomainStackTop(): ScopeLayer | undefined {
+    const stack = this.getDomainStack();
+    if (stack === undefined) {
+      return undefined;
+    }
+    if (stack.length === 0) {
+      stack.push({
+        client: getCurrentClient(),
+        scope: this.getInitialScope(),
+        type: 'domain',
+      });
+    }
+    return stack[stack.length - 1];
+  }
+
+  /**
+   * TODO
+   */
+  private getInitialScope(): any {
+    let initalScope = {};
+    try {
+      initalScope = getCurrentClient() && getCurrentClient().getInitialScope();
+    } catch {
+      // we do nothing
+    }
+    return initalScope;
+  }
+}
+
+/**
+ * TODO
+ */
+function _getLatestShim(): Shim {
```

Comment:
This cannot return `Shim`, as it might be an older or newer version. Just return `any` here